adept/_vlasov1d/storage.py

```diff
@@ -74,43 +75,38 @@ def store_f(cfg: Dict, this_t: Dict, td: str, ys: Dict) -> xr.Dataset:
     return f_store
 
 
-# def clean_td(td):
-#     _ = [os.remove(os.path.join(td, "binary", "fields", fl)) for fl in os.listdir(os.path.join(td, "binary", "fields"))]
-#     _ = [os.remove(os.path.join(td, "binary", "f", fl)) for fl in os.listdir(os.path.join(td, "binary", "f"))]
-#
-#
-# def first_store(td, cfg):
-#     os.makedirs(os.path.join(td, "binary", "f"))
-#     os.makedirs(os.path.join(td, "binary", "fields"))
-#
-#     start_f = jnp.array(cfg["grid"]["f"])
-#     store_f(cfg, np.array([0.0]), td, start_f)
-#     fields = {
-#         "ex": np.zeros((1, cfg["grid"]["nx"], cfg["grid"]["ny"])),
-#         "ey": np.zeros((1, cfg["grid"]["nx"], cfg["grid"]["ny"])),
-#         "total_ex": np.zeros((1, cfg["grid"]["nx"], cfg["grid"]["ny"])),
-#         "total_ey": np.zeros((1, cfg["grid"]["nx"], cfg["grid"]["ny"])),
-#         "dex": np.zeros((1, cfg["grid"]["nx"], cfg["grid"]["ny"])),
-#     }
-#     store_fields(cfg, td, fields, np.array([0.0]))
-#     mlflow.log_artifacts(td)
-#     clean_td(td)
-#
-#     return start_f
-#
-#
-# def store_everything(td, cfg, this_t, fields, this_driver, i, running_f):
-#     fields["dex"] = this_driver[:, 0]
-#     store_fields(cfg, td, fields, this_t)
-#
-#     if i % (cfg["grid"]["num_checkpoints"] // cfg["grid"]["num_fs"]) == 0:
-#         store_f(cfg, this_t[-1:], td, running_f)
-#         mlflow.log_artifacts(td)
-#         clean_td(td)
-
-
-def get_field_save_func(cfg, k):
-    if {"t"} == set(cfg["save"][k].keys()):
+def store_diags(cfg: Dict, this_t: Dict, td: str, ys: Dict) -> xr.Dataset:
+    """
+    Stores f to netcdf
+
+    :param cfg:
+    :param this_t:
+    :param td:
+    :param ys:
+    :return:
+    """
+
+    if {"t"} == set(cfg["save"]["diags"].keys()):
+        axes = (("t", this_t["diag-vlasov"]), ("x", cfg["grid"]["x"]), ("v", cfg["grid"]["v"]))
+    elif {"t", "x", "v"} == set(cfg["save"]["diags"].keys()):
+        axes = (("t", this_t["diag-vlasov"]), ("x", cfg["grid"]["x"]), ("v", cfg["grid"]["v"]))
+    elif {"t", "kx", "v"} == set(cfg["save"]["diags"].keys()):
+        axes = (("t", this_t["diag-vlasov"]), ("kx", cfg["grid"]["kx"]), ("v", cfg["grid"]["v"]))
+    elif {"t", "x", "kv"} == set(cfg["save"]["diags"].keys()):
+        axes = (("t", this_t["diag-vlasov"]), ("x", cfg["grid"]["x"]), ("kv", cfg["grid"]["kv"]))
+    elif {"t", "kx", "kv"} == set(cfg["save"]["diags"].keys()):
+        axes = (("t", this_t["diag-vlasov"]), ("kx", cfg["grid"]["kx"]), ("kv", cfg["grid"]["kv"]))
+    else:
+        raise NotImplementedError
+
+    diags_store = xr.Dataset({spc: xr.DataArray(ys[spc], coords=axes) for spc in ["diag-vlasov", "diag-fp"]})
```

Comment:
You should make this pair of strings, `["diag-vlasov", "diag-fp"]`, an element of the config. That way you could reuse it across the places it appears, and do stuff like `if "diag-vlasov-dfdt" in cfg["diagnostics"]

Reply:
I was hesitant to give each their own save config but it does allow for much cleaner practices inside